### ISSUE TITLE
Output not handling ansi escape code `\u001b[A`

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#291]
Originally posted here https://github.com/microsoft/vscode-python/issues/14476#issuecomment-714895448
Splitting up the issue as there are two issues in the original issue.

**We get the following text stream output (ansi)**
* `Foo\n` (now output should display `foo`)
* `Hello World` (next line we have `Hello World`).
* `\u001B[A` (Ansi escape to move cursor Up)
* `Bar` (as the cursor has moved one line up, we shoudl now get the output
	```
	Bar
	Hello World
	```

* Instead what we get is 
	```
	Foo
	Hello World
	Bar
	```

I.e. the root cause is we do not understand `[A` (move cursor up)

**Here's the real world example (reported in the past as well)**
``
Epoch 0:  92%|█████████▏| 1718/1874 [00:17<00:01, 98.09it/s, loss=2.047, v_num=19, acc=0.875]
Validating: 0it [00:00, ?it/s]
Epoch 0:  92%|█████████▏| 1726/1874 [00:18<00:01, 94.79it/s, loss=2.047, v_num=19, acc=0.875]
Epoch 0:  94%|█████████▎| 1753/1874 [00:18<00:01, 95.74it/s, loss=2.047, v_num=19, acc=0.875]
Epoch 0:  95%|█████████▌| 1782/1874 [00:18<00:00, 96.78it/s, loss=2.047, v_num=19, acc=0.875]
Epoch 0:  97%|█████████▋| 1811/1874 [00:18<00:00, 97.83it/s, loss=2.047, v_num=19, acc=0.875]
Epoch 0:  98%|█████████▊| 1842/1874 [00:18<00:00, 98.96it/s, loss=2.047, v_num=19, acc=0.875]
Epoch 0: 100%|██████████| 1874/1874 [00:18<00:00, 99.57it/s, loss=2.047, v_num=19, acc=0.919] 
Epoch 0: 100%|██████████| 1874/1874 [00:18<00:00, 99.55it/s, loss=2.047, v_num=19, acc=0.919]
```

**When what we should get is:**
Basically when we see `[A` erase the previous line.
```
Epoch 0: 100%|██████████| 1874/1874 [00:18<00:00, 99.55it/s, loss=2.047, v_num=19, acc=0.919]
```

Here's the sample code
```python
import torch
from torch import nn
from torch.nn import functional as F

from pl_bolts.datamodules.mnist_datamodule import MNISTDataModule
import pytorch_lightning as pl
class LitLeNet(pl.LightningModule):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        self.conv1 = nn.Conv2d(1, 6, 5, padding=2)
        self.conv2 = nn.Conv2d(6, 16, 5)
        self.fc1 = nn.Linear(16 * 5 * 5, 120)
        self.fc2 = nn.Linear(120, 84)
        self.classifier = nn.Linear(84, 10)

        self.accuracy = pl.metrics.Accuracy()
    
    def forward(self, x):
        x = F.relu(self.conv1(x))
        x = F.max_pool2d(x, (2, 2))
        x = F.relu(self.conv2(x))
        x = F.max_pool2d(x, (2, 2))
        x = x.view(-1, 16 * 5 * 5)
        x = F.relu(self.fc1(x))
        x = F.relu(self.fc2(x))
        x = self.classifier(x)
        x = F.softmax(x, dim=0)

        return x
    
    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters())
    
    def _shared_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        self.log('acc', self.accuracy(y_hat, y), prog_bar=True)
        return loss

    def training_step(self, batch, batch_idx):
        return self._shared_step(batch, batch_idx)
    
    def validation_step(self, batch, batch_idx):
        return self._shared_step(batch, batch_idx)
    
    def test_step(self, batch, batch_idx):
        return self._shared_step(batch, batch_idx)
    
    def validation_epoch_end(self, outputs):
        self.log('acc', self.accuracy.compute(), prog_bar=True)
mnist = MNISTDataModule('./data', num_workers=2)
model = LitLeNet()

trainer = pl.Trainer(max_epochs=1)
trainer.fit(model, datamodule=mnist)
```

**Comment by DonJayamanne** [comment]
This has been fixed
